### PR TITLE
Feature hourly and quote based bare metal server

### PR DESCRIPTION
**Pull request by minsikl**
I cloned the PR #151 and extended the bare metal resource.

The following functions are added. 
Added hourly bare metal server provisioning functions.
* CPU, Memory, Disk and RAID configuration functions

Added quote based bare metal server provisioning functions.

Added quote_bare_metal data source.
* Data source which retrieves hourly bare metal server order information.

### CODE REVIEW

File: softlayer/resource_softlayer_bare_metal.go [hunk 1]
```diff
@@ -76,100 +144,164 @@ func resourceSoftLayerBareMetal() *schema.Resource {
...
 
-			"user_metadata": {
-				Type:     schema.TypeString,
+			// Quote based provisioning only
+			"quote_id": {
+				Type:             schema.TypeInt,
+				Optional:         true,
+				ForceNew:         true,
+				DiffSuppressFunc: applyOnce,
+			},
```
Comment (renier) [inline softlayer/resource_softlayer_bare_metal.go:263]: I thought we were going to rely on information from the quote data source to fill in properties in the compute resource. Can you explain why we need the quote_id here?

Comment (minsikl) [reply softlayer/resource_softlayer_bare_metal.go:263]: You can create a bare metal server with the quote_id attribute.

Comment (renier) [reply softlayer/resource_softlayer_bare_metal.go:263]: Yes, so they can't use the data source for this? If not, what is the value-add of the quote data source?

Comment (minsikl) [reply softlayer/resource_softlayer_bare_metal.go:263]: Quote data source provides required attributes for bare metal provisioning. You can refer to the quote data source to define a bare metal server resource. Or you can use interpolations to define a bare metal server resource. For example, 
```hcl
package_key_name = "${data.softlayer_quote_bare_metal.bm1.package_key_name}"
...
```

If you want to define a bare metal server resource with a quote data, you can write a definition as follows:
```hcl
data softlayer_quote_bare_metal quote_bm1{
  name = "quote1"
}

resource softlayer_bare_metal bm1 {
...
quote_id = "${data.softlayer_quote_bare_metal.quote_bm1.id}"
...
}
```

Comment (renier) [reply softlayer/resource_softlayer_bare_metal.go:263]: I don't see why we would need both a quote id and a quote data source. Let's talk about that.

Comment (minsikl) [reply softlayer/resource_softlayer_bare_metal.go:263]: Per your request, the quote_bare_metal data source was deleted.

**Pull request merged**